cub/cub/util_arch.cuh

```diff
@@ -112,28 +114,61 @@ inline constexpr bool prefer_conflict_over_padding = CUB_PTX_PREFER_CONFLICT_OVE
 // Note that in contrast to dynamic shared memory, static shared memory is still limited to 48 KB
 static constexpr ::cuda::std::size_t max_smem_per_block = 48 * 1024;
 
+struct scaling_result
+{
+  int items_per_thread;
+  int block_threads;
+};
+
+_CCCL_API constexpr auto
+scale_reg_bound(int nominal_4B_block_threads, int nominal_4B_items_per_thread, int target_type_size) -> scaling_result
```

Comment:
```suggestion
[[nodiscard]] _CCCL_API inline constexpr auto
scale_reg_bound(int nominal_4B_block_threads, int nominal_4B_items_per_thread, int target_type_size) -> scaling_result
```

Reply:
Is that really meaningful? `inline` is implied by `constexpr` and since the function has no side effects, not using the return value is pointless.